avro-fastserde/src/main/java/com/linkedin/avro/fastserde/primitive/PrimitiveBooleanArrayList.java

```diff
@@ -0,0 +1,87 @@
+package com.linkedin.avro.fastserde.primitive;
+
+import com.linkedin.avro.api.PrimitiveBooleanList;
+import org.apache.avro.Schema;
+
+
+public class PrimitiveBooleanArrayList extends PrimitiveArrayList<Boolean, PrimitiveBooleanList> implements PrimitiveBooleanList {
+  public static final Schema SCHEMA = Schema.createArray(Schema.create(Schema.Type.BOOLEAN));
+  boolean[] elementsArray;
+
+  public PrimitiveBooleanArrayList(int capacity) {
+    super(PrimitiveBooleanList.class);
+    this.elementsArray = new boolean[capacity];
+  }
+
+  @Override
+  public Boolean get(int index) {
+    return getPrimitive(index);
+  }
+
+  @Override
+  public boolean getPrimitive(int index) {
+    checkIfLargerThanSize(index);
```

Comment:
woudlnt java naturally do this ?

Reply:
No it wouldn't. The `size` property represents the boundary of relevant elements within the potentially larger `elementsArray`. The array is allowed to be larger because of instance reuse, where we wouldn't shrink the array if we need fewer elements than it has capacity for. I will leave some JavaDoc to indicate this.